### PR TITLE
refactor(ruby): integrating general db schema in the service

### PR DESCRIPTION
**Pull request by dudustri**
- makes ruby service compatible with the main DB schema!

### CODE REVIEW

File: ruby-sinatra/README.md [hunk 1]
```diff
@@ -13,6 +13,6 @@ sudo apt update && sudo apt install libpq-dev
 ### Setup
 
 - Install required gems with `bundle install`
-- Initial database with `bundle exec rake db:create`
-- Run migrations `bundle exec rake db:migrate`
+- Initial database with `bundle exec rake db:create` - If the DB doesn't Exists!!!
+- Run migrations `bundle exec rake db:migrate` - Not needed!!
```
Comment (Ghmeding) [inline ruby-sinatra/README.md:17]: Why do we have it then?

File: ruby-sinatra/README.md [hunk 2]
```diff
@@ -13,6 +13,6 @@ sudo apt update && sudo apt install libpq-dev
 ### Setup
 
 - Install required gems with `bundle install`
-- Initial database with `bundle exec rake db:create`
-- Run migrations `bundle exec rake db:migrate`
+- Initial database with `bundle exec rake db:create` - If the DB doesn't Exists!!!
```
Comment (Ghmeding) [inline ruby-sinatra/README.md:16]: I am not sure I understand this. When should this command be run? When the db-container is running or?

**Review by Ghmeding — COMMENTED** [review]
Really nice work! <3

The only thing is that I am having issues running it on my side.

I installed bundler and ruby 3.1.0. I ran `bundle install` successfully and also span the db-container up. 

When I run `bundle exec ruby myapp.rb` the service spins up but when I get an error when I try to get: 
_"ActiveRecord::NoDatabaseError - We could not find your database: waect. Available database configurations can be found in config/database.yml."_

I also tried running `bundle exec rake db:create` but then I get this:
_"No Rakefile found (looking for: rakefile, Rakefile, rakefile.rb, Rakefile.rb)"_

File: ruby-sinatra/README.md (hunk 2 above)
Comment (dudustri) [reply ruby-sinatra/README.md:16]: If our db container is running it exists heheh So you dont need to run it.

File: ruby-sinatra/README.md (hunk 1 above)
Comment (dudustri) [reply ruby-sinatra/README.md:17]: Because refactor is not the scope of this PR

Comment (Ghmeding) [reply ruby-sinatra/README.md:17]: Should we write it as a TODO then?

File: ruby-sinatra/README.md (hunk 2 above)
Comment (dudustri) [reply ruby-sinatra/README.md]: I will delete it.

**Comment by dudustri** [comment]
Check your IP or try to use localhost instead when running. Also, check the bundler and ruby version installed. Probably you have already but double check the libpq-dev library.

File: ruby-sinatra/README.md (hunk 1 above)
Comment (dudustri) [reply ruby-sinatra/README.md]: No, I just deleted because we are not using it at all

File: ruby-sinatra/myapp.rb [hunk 3]
```diff
@@ -25,32 +32,24 @@
...
-      user: db.user,
-      password: db.password,
-      encoding: "utf8",
-    }
+    set :database, DATABASE_URL
   else
-    set :database, { adapter: "postgresql", database: "minitwit_test" }
+    # Fall back to specific configurations for development and test environments
+    case settings.environment
+    when :development
```
Comment (simoonH) [inline ruby-sinatra/myapp.rb:42]: Maybe we can add the specified varibales here and comment out the host so we can run it locally. Depending on, if we're just using the image in the future
`{ adapter: "postgresql",
            database: "waect" ,
            host: "192.168.8.175",
            port: 5432,
            database: "waect",
            username: "user",
            password: "pass"
      }`

File: README.md [hunk 4]
```diff
@@ -1,3 +1,17 @@
 # codebase
 
 WAEC Project Code Base
+
```
Comment (simoonH) [inline README.md:4]: Nice! Will include this in the c# repo as well

File: ruby-sinatra/simapi/sim_api.rb [hunk 5]
```diff
@@ -4,10 +4,16 @@
 require "sinatra"
 require "json"
 require "newrelic_rpm"
+require 'dotenv'
+
```
Comment (simoonH) [inline ruby-sinatra/simapi/sim_api.rb:8]: Not your fault, but something to consider for the future:
* We might want to rename this folder, as the current naming is not very clear. Something like `api` or `controllers` could be better?
* We could also consider separating the routes into different controllers, for example, a controller for users, another for messages, etc. - think it's convention for sinatra and ruby but don't know 100%

**Review by simoonH — APPROVED** [review]
Nice work man! Looks good to me, nothing major, just some little things to keep in mind for the future. :)

File: ruby-sinatra/simapi/sim_api.rb [hunk 6]
```diff
@@ -27,32 +33,24 @@
...
-      user: db.user,
-      password: db.password,
-      encoding: "utf8",
-    }
+    set :database, DATABASE_URL
   else
-    set :database, { adapter: "postgresql", database: "minitwit_test" }
+    # Fall back to specific configurations for development and test environments
+    case settings.environment
+    when :development
```
[the remark this answers is not in the document]
Comment (dudustri) [reply ruby-sinatra/simapi/sim_api.rb:42]: We should change this approach entirely for sure. To run the integration tests we can just use the same DB and delete it afterwards, or spin a DB-test container without volume and just stop it after the tests. Something similar with what we have already.

**Pull request merged**